src/blueapi/cli/cli.py

```diff
@@ -33,23 +33,16 @@
 )
 @click.pass_context
 def main(ctx: click.Context, config: Path | None | tuple[Path, ...]) -> None:
-    # if no command is supplied, run with the options passed
-
-    config_loader = ConfigLoader(ApplicationConfig)
-    if config is not None:
-        configs = (config,) if isinstance(config, Path) else config
-        for path in configs:
-            if path.exists():
-                config_loader.use_values_from_yaml(path)
-            else:
-                raise FileNotFoundError(f"Cannot find file: {path}")
-
-    ctx.ensure_object(dict)
-    loaded_config: ApplicationConfig = config_loader.load()
-
-    ctx.obj["config"] = loaded_config
+    # Override default yaml_file path in the model_config if `config` is provided
+    ApplicationConfig.model_config["yaml_file"] = config
```

Comment:
Does this handle the case of `tuple[Path, ...]`?

Reply:
not sure, not sure if we have a test for it at the moment.

might need to add some logic to handle that

Reply:
`pydantic_settings` defines PathType as: `PathType = Union[Path, str, Sequence[Union[Path, str]]]`